org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -110,6 +112,13 @@ import org.eclipse.xtext.validation.Issue
 		if (params.rootPath === null) {
 			throw new IllegalArgumentException("Bad initialization request. rootPath must not be null.")
 		}
+		val rootPath = Paths.get(params.rootPath)
+		if (!Files.exists(rootPath) || !Files.isDirectory(rootPath)) {
+			throw new IllegalArgumentException("Bad initialization request. rootPath does not exist or is not a directory.")
```

Comment:
Could we make two separate checks instead? I think it would be nice to see what the exact problem was.
So something like this:
```
if (!Files.exists(rootPath)) {
    throw new IAE('does not exist');
}
if (!Files.isDirectory(rootPath)) {
    throw new IAE('not a directory');
}
```